File: scola-api/app/models/attendance.rb
class Attendance < ApplicationRecord
  belongs_to :lesson
  belongs_to :student

  enum :status, { present: 'present', absent: 'absent', late: 'late', excused: 'excused' }

  validates :date, presence: true
  validates :status, presence: true
  validates :student_id, uniqueness: { scope: [:lesson_id, :date] }
end

File: scola-app/src/types/attendance.ts
export type AttendanceStatus = 'present' | 'absent' | 'late' | 'excused';

export interface Attendance {
  id: number;
  student_id: number;
  student_name: string;
  date: string;
  status: AttendanceStatus;
  note: string | null;
}
